compiler/modules/bir/verify.bal

```diff
@@ -151,6 +216,12 @@ function verifyInsn(VerifyContext vc, Insn insn) returns Error? {
     else if insn is ErrorConstructInsn {
         check validOperandString(vc, name, insn.operand, insn.pos);
     }
+    else if insn is BranchInsn && insn.backward {
+        if check terminator(vc, vc.getBlock(insn.dest), insn.pos) is BranchInsn|CondBranchInsn|TypeBranchInsn
```

Comment:
This is checking that your can't `return` form the LoopHead. I don't think this is a useful thing to check. What we need is a much stronger guarantee. That is, LoopHead is to a dominator. This will automatically guarantee that LoopHead can't `return`.